contracts/DegenLockToken.sol

```diff
@@ -0,0 +1,194 @@
+
+    mapping(address account => uint256) private _depositTimestamps;
+
+    /**
+     * @dev Name of the token representing the claim on the locked token
+     */
+    string public constant TOKEN_NAME = "Locked Degen";
+
+    /**
+     * @dev Symbol of the token representing the claim on the locked token
+     */
+    string public constant TOKEN_SYMBOL = "LDEGEN";
+
+    /**
+     * @dev The ERC20 token to be locked
+     */
+    address public constant TOKEN = 0xfee293840D23B0B2De8C55e1Cf7A9F01C157767c;
+
+    /**
+     * @dev Lock duration in seconds, period starts after the deposit timestamp
+     */
+    uint256 public lockDuration = 90 days;
+
+    /**
+     * @dev Minimum amount of tokens that have to be deposited
+     */
+    uint256 public minDepositAmount;
+
+    /**
+     * @dev Withdraw is not possible because the lock period is not over yet
+     */
+    error LockPeriodOngoing();
+
+    /**
+     * @dev ERC-20 function is not supported
+     */
+    error NotSupported();
+
+    /**
+     *  @dev The lock duration is longer than a year
+     */
+    error LockDurationIsTooLong();
+
+    /**
+     *  @dev Amount of tokens to be locked is zero
+     */
+    error ZeroAmount();
+
+    /**
+    /**
+     * @dev Minimum deposit not met
+     */
+    error MinimumDepositNotMet();
+
+    /**
+     *  @dev The lock duration has been updated
+     */
+    event LockDurationUpdated(uint256 indexed duration);
+
+    /**
+     *  @dev The minimum deposit amount has been updated
+     */
+    event MinDepositAmountUpdated(uint256 indexed minDepositAmount);
+
+    /**
+     *  @dev The deposit timestamp has been set
+     */
+    event DepositTimestampUpdated(
+        address indexed account,
+        uint256 indexed depositTimestamp
+    );
+
+    /**
+     * @dev Construct a new Degen token
+     */
+    constructor() ERC20(TOKEN_NAME, TOKEN_SYMBOL) Ownable(msg.sender) {
+        minDepositAmount = 10000 * 10 ** decimals();
+        emit MinDepositAmountUpdated(minDepositAmount);
+    }
+
+    /**
+     * @dev Deposit tokens to be locked until the end of the locking period
+     * @param amount The amount of tokens to deposit
+     */
+    function deposit(uint256 amount) external nonReentrant {
+        if (amount < minDepositAmount) {
+            revert MinimumDepositNotMet();
+        }
+
+        IERC20(TOKEN).safeTransferFrom(msg.sender, address(this), amount);
+        _mint(msg.sender, amount);
+
+        _depositTimestamps[msg.sender] = block.timestamp;
+        emit DepositTimestampUpdated(msg.sender, block.timestamp);
+    }
+
+    /**
+     * @dev Withdraw tokens after the end of the locking period or during the deposit period
+     * @param amount The amount of tokens to withdraw
+     */
+    function withdraw(uint256 amount) external nonReentrant {
+        if (amount == 0) {
+            revert ZeroAmount();
+        }
+
+        if (block.timestamp < _depositTimestamps[msg.sender] + lockDuration) {
```

Comment:
I think `<=` might be the correct operator since `block.timestamp` should exceed the the lock period (i.e. `_depositTimestamps[msg.sender] + lockDuration`).

Reply:
Fixed.